app/app/explore/components/AddLiquidityModal.tsx

```diff
@@ -39,24 +37,29 @@ export default function AddLiquidityModal({
   onClose,
 }: AddLiquidityModalProps) {
   const inputRef = useRef<HTMLInputElement>(null);
-  const { error: liquidityError, addLiquidity } = useLiquidityPosition({
+  const {
+    error: liquidityError,
+    addLiquidity,
+    getAssetWallet,
+  } = useLiquidityPosition({
     pool,
   });
-  const { toggleWalletModal, walletsState, balanceList } = useAppState();
-  const { getNetworkAddressFromLocalStorage, hasThorAddressInLocalStorage } =
-    useWalletConnection();
+  const { toggleWalletModal, walletsState, balanceList, isWalletConnected } =
+    useAppState();
 
-  // Parse asset details
   const [assetChain] = useMemo(
     () => parseAssetString(pool.asset),
     [pool.asset],
   );
-  const chainKey = getChainKeyFromChain(assetChain);
-  const selectedWallet =
-    walletsState && walletsState[chainKey] ? walletsState[chainKey] : null;
+  const chainKey = useMemo(
+    () => getChainKeyFromChain(assetChain),
+    [assetChain],
+  );
```

Comment:
I wonder if in this cases the use of `useMemo` is really an optimisation since `getChainKeyFromChain` is not computationally expensive. I have been consulting a bit and I think that maybe we should limit the use of `useMemo` for those cases where the computation cost is constant or the update frequency is high. An excessive use can be counterproductive, here I leave one of the articles that led me to this conclusion to see what you think

https://dev.to/ayoashy/understanding-reacts-usememo-what-it-does-when-to-use-it-and-best-practices-4gdh?utm_source=chatgpt.com

Do you think these useMemo are really necessary?